File: apps/frontend/src/types/request.ts
import type { BloodComponent, BloodType, RequestStatus, RequestUrgency } from "@/types/domain";

export type { RequestStatus, RequestUrgency } from "@/types/domain";

export interface BloodRequest {
  id: string;
  request_number: string;
  hospital_id: string;
  blood_type: BloodType;
  component: BloodComponent;
  units_requested: number;
  urgency: RequestUrgency;
  status: RequestStatus;
  requested_at: string;
  fulfilled_at: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

File: apps/backend/app/schemas/request.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import BloodComponent, BloodType, RequestStatus, RequestUrgency


class BloodRequestCreate(BaseModel):
    request_number: str | None = Field(default=None, min_length=3, max_length=50)
    hospital_id: UUID
    blood_type: BloodType
    component: BloodComponent
    units_requested: int = Field(gt=0)
    urgency: RequestUrgency
    notes: str | None = None


class BloodRequestUpdate(BaseModel):
    blood_type: BloodType | None = None
    component: BloodComponent | None = None
    units_requested: int | None = Field(default=None, gt=0)
    urgency: RequestUrgency | None = None
    status: RequestStatus | None = None
    notes: str | None = None


class BloodRequestRead(BaseModel):
    id: UUID
    request_number: str
    hospital_id: UUID
    blood_type: BloodType
    component: BloodComponent
    units_requested: int
    urgency: RequestUrgency
    status: RequestStatus
    requested_at: datetime
    fulfilled_at: datetime | None
    notes: str | None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
